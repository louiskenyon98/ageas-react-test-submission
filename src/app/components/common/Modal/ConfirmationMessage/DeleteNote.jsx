import React from 'react';
import ConfirmationMessage from "./ConfirmationMessage";

export default class DeleteNote extends React.PureComponent {
    render() {
        return (
            <ConfirmationMessage
                title={"Are you sure you want to delete your note"}
                body={"Hello world"}
                decline={{
                    label: "CANCEL",
                    onClick: this.props.cancel
                }}
                accept={{
                    label: "DELETE",
                    onClick: this.props.delete
                }}
            />
        )
    }
}
