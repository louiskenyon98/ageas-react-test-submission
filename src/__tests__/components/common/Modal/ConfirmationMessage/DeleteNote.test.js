import React from 'react';
import {shallow} from 'enzyme';
import DeleteNote from '../../../../../app/components/common/Modal/ConfirmationMessage/DeleteNote';

describe('DeleteNote', () => {
    let props = {};
    let wrapper = () => shallow(<DeleteNote {...props}/>);
    beforeEach(() => {
        props = {
            cancel: jest.fn(),
            delete: jest.fn()
        }
    });
    describe('render', () => {
        it('should render ConfirmationMessage with title, body, and should have accept/decline objects', () => {
            expect(wrapper()).toMatchSnapshot();
        })
    })
});
