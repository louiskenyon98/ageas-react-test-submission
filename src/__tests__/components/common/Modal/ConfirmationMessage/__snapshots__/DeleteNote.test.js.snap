// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteNote render should render ConfirmationMessage with title, body 1`] = `
<ConfirmationMessage
  accept={
    Object {
      "label": "DELETE",
      "onClick": [MockFunction],
    }
  }
  body="Hello world"
  decline={
    Object {
      "label": "CANCEL",
      "onClick": [MockFunction],
    }
  }
  title="Are you sure you want to delete your note"
/>
`;
